src/frontend/src/scheduler/distributed/query_manager.rs

```diff
@@ -94,13 +99,13 @@ impl QueryManager {
             .get_epoch(query_id.clone())
             .await?;
 
-        let creat_task_resp = compute_client
+        compute_client
             .create_task(task_id.clone(), plan, epoch)
-            .await;
+            .await
+            .map_err(|e| anyhow!(e))?;
         self.hummock_snapshot_manager
             .unpin_snapshot(epoch, &query_id)
             .await?;
-        creat_task_resp?;
```

Comment:
Is this by design to first unpin the snapshot and then throw the potential error?

Reply:
Why we need to explicitly unpin a snapshot? We should have something like `SnapshotHandler`, and when dropping the handler, the snapshot gets automatically unpinned.

Reply:
Indeed the best way is not to explicitly call pin/unpin. We should wrap the pin/unpin in a query's new() and drop(). 

But backs to current quick answer, it is becasue we do not have a error handle of schedule a DML task (only task that still use schedule_single). So we have to manually unpin the snapshot if we do not want the unpin be blocked...